src/cli/repl/meta_cmd.rs

```diff
@@ -649,8 +669,503 @@ impl<F: LurkField + DeserializeOwned> MetaCmd<F> {
+        name: "dump-data",
+        summary: "Write Lurk data to the file system",
+        format: "!(dump-data <expr> <string>)",
+        description: &[],
+        example: &["!(dump-data (+ 1 1) \"my_file\")"],
+        run: |repl, args| {
+            let (expr, path) = repl.peek2(args)?;
+            let path = get_path(repl, &path)?;
+            let (io, ..) = repl
+                .eval_expr(expr)
+                .with_context(|| "evaluating predicate")?;
+            let mut z_dag = ZDag::default();
+            let z_ptr = z_dag.populate_with(&io[0], &repl.store, &mut Default::default());
+            dump(LurkData { z_ptr, z_dag }, &path)
+        },
+    };
+}
+
+impl<F: LurkField + DeserializeOwned> MetaCmd<F> {
+    const LOAD_DATA: MetaCmd<F> = MetaCmd {
+        name: "load-data",
+        summary: "Read Lurk data from the file system and bind it to a symbol",
+        format: "!(load-data <symbol> <string>)",
+        description: &[],
+        example: &["!(load-data x \"my_file\")"],
+        run: |repl, args| {
+            let (sym, path) = repl.peek2(args)?;
+            if !matches!(sym.tag(), Tag::Expr(ExprTag::Sym)) {
+                bail!(
+                    "Bound variable must be a symbol. Got {}",
+                    sym.fmt_to_string(&repl.store, &repl.state.borrow())
+                )
+            }
+            let path = get_path(repl, &path)?;
+            let LurkData::<F> { z_ptr, z_dag } = load(&path)?;
+            let ptr = z_dag.populate_store(&z_ptr, &repl.store, &mut Default::default())?;
+            let binding = repl.store.cons(sym, ptr);
+            repl.env = repl.store.cons(binding, repl.env);
+            Ok(())
+        },
+    };
+}
+
+impl MetaCmd<F> {
+    const DEFPROTOCOL: MetaCmd<F> = MetaCmd {
+        name: "defprotocol",
+        summary: "Defines a protocol",
+        format: "!(defprotocol <symbol> <vars> <body> options...)",
+        description: &[
+            "The protocol body cannot have any free variable besides the ones",
+            "  declared in the vars list. The body must return a pair such that:",
+            "* The first component is a (proper) list containing the CEK inputs",
+            "  that the proof must accept. This list has 6 elements and follow",
+            "  the order (expr-in env-in cont-in expr-out env-out cont-out).",
+            "  The protocol can deny the proof by returning nil instead of a list.",
+            "* The second component is a 0-arg predicate that will run after the",
+            "  proof verification to further constrain the proof, if needed.",
+            "  If this is not necessary, this component can simply be nil instead.",
+            "",
+            "defprotocol accepts the following options:",
+            "  :rc is the reduction count, defaulting to the REPL's one",
+            "  :lang defines the Lang (ignored for now)",
+            "  :descr is a description of the protocol, defaulting to \"\""
```

Comment:
`:descr` can and should just be `:description`. Symbols are not charged by the character, and keywords should be explicit and descriptive. `:lang` and `:rc` are okay since those are 'technical terms' as used in Lurk.